packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/BarcodeDetector.java

```diff
@@ -115,64 +118,68 @@ public void onSuccess(List<FirebaseVisionBarcode> firebaseVisionBarcodes) {
                       }
                       typeValue.put("emails", emails);
 
-                      Map<String, Object> name = new HashMap();
-                      name.put(
-                          "formatted_name", barcode.getContactInfo().getName().getFormattedName());
-                      name.put("first", barcode.getContactInfo().getName().getFirst());
-                      name.put("last", barcode.getContactInfo().getName().getLast());
-                      name.put("middle", barcode.getContactInfo().getName().getMiddle());
-                      name.put("prefix", barcode.getContactInfo().getName().getPrefix());
-                      name.put(
-                          "pronunciation", barcode.getContactInfo().getName().getPronunciation());
-                      name.put("suffix", barcode.getContactInfo().getName().getSuffix());
+                      Map<String, Object> name = new HashMap<>();
+                      if (barcode.getContactInfo().getName() != null) {
```

Comment:
Maybe save `barcode.getContactInfo().getName()` in an instance variable to make the code below simpler?

Reply:
Might also be nice for the code below...